internal/util/util.go

```diff
@@ -423,3 +428,27 @@ func Sample(m proto.Message, strategy *SamplingStrategy) proto.Message {
 	})
 	return m
 }
+
+// HashStore is a simple a store to produce hash for protobuf message
+type HashStore struct {
+	store map[string]int
+}
+
+// GetHash retrieves a hash string for a given protobuf message.
+// Note this should be restrict to a request scope.
+func (hs *HashStore) GetHash(m proto.Message) string {
+	if hs.store == nil {
+		hs.store = map[string]int{}
+	}
+	b, err := proto.Marshal(m)
```

Comment:
Should we worry about the stability of using serialized generic proto as a hash key, esp in the presence of map fields? given https://stackoverflow.com/a/60512308

If this is only needed for metadata for now, I wonder about making it specific to that.

Reply:
Changed this to hash the concatenated StatMetadata field strings now.